eli5/keras/explain_prediction.py

```diff
@@ -0,0 +1,226 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from typing import Union, Optional, Callable, Tuple, List
+
+import numpy as np # type: ignore
+import keras # type: ignore
+import keras.backend as K # type: ignore
+from keras.models import Model # type: ignore
+from keras.layers import Layer # type: ignore
+
+from eli5.base import Explanation, TargetExplanation
+from eli5.explain import explain_prediction
+from .gradcam import gradcam, gradcam_backend
+
+
+DESCRIPTION_KERAS = """Grad-CAM visualization for image classification; 
+output is explanation object that contains input image 
+and heatmap image for a target.
+"""
+
+# note that keras.models.Sequential subclasses keras.models.Model
+@explain_prediction.register(Model)
+def explain_prediction_keras(estimator, # type: Model
+                             doc, # type: np.ndarray
+                             target_names=None,
+                             targets=None, # type: Optional[list]
+                             layer=None, # type: Optional[Union[int, str, Layer]]
+                            ):
+    # type: (...) -> Explanation
+    """
+    Explain the prediction of a Keras image classifier.
+
+    We make two explicit assumptions
+        * The input is images.
+        * The model's task is classification, i.e. final output is class scores.
+
+    See :func:`eli5.explain_prediction` for more information about the ``estimator``,
+    ``doc``, ``target_names``, and ``targets`` parameters.
+
+    
+    :param keras.models.Model estimator:
+        Instance of a Keras neural network model, 
+        whose predictions are to be explained.
+
+
+    :param numpy.ndarray doc:
+        An input image as a tensor to ``estimator``, 
+        from which prediction will be done and explained.
+
+        For example a ``numpy.ndarray``.
```

Comment:
are there other supported data types, why "for example"?

Reply:
I will explicitly mention that numpy arrays are required.

I think it is possible to use other input types (https://github.com/keras-team/keras/blob/ed07472bc5fc985982db355135d37059a1f887a9/keras/engine/training.py#L1315), i.e. tensorflow tensor. However, I haven't tested with other types and I think I have some numpy dependencies in my code. Adding more input types could be a separate GitHub issue?

https://github.com/TeamHG-Memex/eli5/pull/315/commits/9d2d22af443be5b96a1d0626b78943752023cb07